storage/index.go

```diff
@@ -0,0 +1,105 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package storage
+
+type searchIndexAll struct {
+	Packages []packageIndex `json:"packages"`
+}
+
+type packageIndex struct {
+	PackageManifest     packageManifest      `json:"package"`
+	DataStreamManifests []dataStreamManifest `json:"data_streams"`
+
+	Assets []string `json:"assets"`
+}
+
+type packageManifest struct {
+	FormatVersion string  `json:"format_version,omitempty"`
+	Name          string  `json:"name,omitempty"`
+	Title         string  `json:"title,omitempty"`
+	Version       string  `json:"version,omitempty"`
+	Release       string  `json:"release,omitempty"`
+	License       string  `json:"license,omitempty"`
+	Description   string  `json:"description,omitempty"`
+	Type          string  `json:"type,omitempty"`
+	Icons         []image `json:"icons,omitempty"`
+	Screenshots   []image `json:"screenshots,omitempty"`
+	Conditions    *struct {
+		Kibana *struct {
+			Version string `json:"version,omitempty"`
+		} `json:"kibana,omitempty"`
+	} `json:"conditions,omitempty"`
+	Owner *struct {
+		Github string `json:"github,omitempty"`
+	} `json:"owner,omitempty"`
+	Categories []string `json:"categories,omitempty"`
+
+	PolicyTemplates []struct {
+		Name        string   `json:"name,omitempty"`
+		Title       string   `json:"title,omitempty"`
+		Categories  []string `json:"categories,omitempty"`
+		DataStreams []string `json:"data_streams,omitempty"`
+		Description string   `json:"description,omitempty"`
+		Icons       []image  `json:"icons,omitempty"`
+		Input       struct {
+			Title        string     `json:"title,omitempty"`
+			Type         string     `json:"type,omitempty"`
+			Description  string     `json:"description,omitempty"`
+			InputGroup   string     `json:"input_group,omitempty"`
+			TemplatePath string     `json:"template_path,omitempty"`
+			Vars         []variable `json:"vars,omitempty"`
+		} `json:"input,omitempty"`
+		Screenshots []image    `json:"screenshots,omitempty"`
+		Vars        []variable `json:"vars,omitempty"`
+	} `json:"policy_templates,omitempty"`
+}
```

Comment:
Oh no, another place where we have to keep these structs :sweat_smile:  https://github.com/elastic/package-spec/issues/322

Reply:
I would like to see something generated from the spec, like https://github.com/elastic/package-spec/issues/322.

Reply:
It is slightly different from the spec as it includes extra properties like assets and data streams hence I was thinking about a dedicated repo for the model.

Reply:
Ah, yes, you are right, this is not the same.